pkg/sdk/go/message.go

```diff
@@ -69,3 +71,135 @@ func (sdk *mgSDK) SetContentType(ct ContentType) errors.SDKError {
 
 	return nil
 }
+
+func (sdk mgSDK) withMessageQueryParams(baseURL, endpoint string, mpm MessagePageMeta) (string, error) {
+	q, err := mpm.mquery()
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%s/%s?%s", baseURL, endpoint, q), nil
+}
+
+func (mpm MessagePageMeta) mquery() (string, error) {
```

Comment:
This looks like it can be handled better with a map and JSON serialization of MessagePageMeta.